logrotate.c

```diff
@@ -2554,6 +2554,11 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    /* Creates the dir for state file and moves old
+       state file to new location */
+    mkdir(STATEDIR, 0755);
+    rename(OLDSTATEFILE, STATEFILE);
+
```

Comment:
> imho the change of statefile path should be handled by downstream distribution, and they 
> should provide appropriate upgrade path

All others projects have their on dir in /var/lib, why logrotate want to be different?

`$ ls /var/lib/
abrt             dav        libvirt         os-prober       stateless
AccountsService  dbus       lightdm         plymouth        systemd
alsa             dhclient   lightdm-data    polkit-1        tpm
alternatives     dnf        lldpad          pulse           udisks2
authconfig       dnsmasq    logrotate       rpm             unbound
blueman          fprint     machines        rpm-state       upower
bluetooth        games      misc            rsyslog         xkb
chrony           geoclue    mlocate         samba           yum
color            gssproxy   net-snmp        selinux
colord           hyperv     NetworkManager  setroubleshoot
corosync         initramfs  nfs             sheepdog
cs               iscsi      openvswitch     sss
`

> 
> example, for rpm:
> %triggerpostun -- logrotate < 3.10.1
> mv -f /var/run/logrotate.status /var/run/logrotate/logrotate.status || :

I am favor of minimum changes in the spec for the statefile PR. So this change could be transparent to others distros.